multi-tenancy-observability/README.md

```diff
@@ -0,0 +1,270 @@
+
+But this approach brings a few questions:
+1. What tenancy should common internal apps use (e.g. prometheus-operator, api-server, cilium logs are accessed by multiple internal teams)? Do we need to ensure customers are putting those apps into a namespace accessible by all?
+2. What about teams that share namespaces?
+
+##### App == Tenant
+
+This approach is especially flexible for customers but we do not think this approach makes sense because the mapping configuration would be impossible to maintain and we would probably face technical limits pretty fast
+
+##### Hybrid mode
+
+This approach would allow customers to be able to set the tenancy as they want (per namespace or per app depending on the use case).
+
+#### Implementation details
+
+The implementation details are at this stage are still an open topic, because no single tool fulfills all of our requirements. Until this RFC we use two tools, grafana-agent and promtail, which are bot not yet feature complete.
+One preferred alternative would be [PodLogs](https://grafana.com/docs/agent/latest/operator/api/#podlogs-a-namemonitoringgrafanacomv1alpha1podlogsa), the equivalent of a Service or PodMonitor for metrics, but the existing implementation at the time of this RFC is not yet fully compliant with our needs as well.
+
+We envisionned different distinct approaches here:
+
+#### Approach 1: Customers provide their logs themselves
+
+In this approach, based on the tenant information that the customer provided to us, we would generate a user/password pair that the customer would then use in their favorite log shipper to send their logs to Loki.
+
+Pros:
+- This is quite easy for us to implement as we already have the password generation logic
+- We allow customers to use the tool they know and love.
+- We will not receive pages if the tool is not working properly
+
+Cons:
+- This is far from the idea of a platform
+- There is a high risk that configuration will be messy and hard for use to debug when we need to
+
+This approach might be useful for customers that want to send logs from outside the clusters
+
+#### Approach 2: Extract tenant information from a label/PodLog
+
+Existing logging agents support setting the tenant to send to loki using a custom log label.
+We can configure our logging agents to get the logs of specific namespaces/pods if they have a label like `giantswarm.io/tenant: my-tenant` or using a PodLog CR (if applicable).
+
+Pros:
+- Quite easy to configure for logs
+
+Cons:
+- Tenant configuration will need to be configured in multiple places (MC for the read path and WC for the write path)
+- Label approach would most likely not work for metrics OOTB (this would require customer to add new labels like metrics port, path and so on).
+
+#### Approach 3: Configure promtail based on the MC tenancy CR
```

Comment:
Unless the community provides a better option, some Giant Swarm abstraction is the likely end state for this use case. BUT, whether it is WC or MC level depends on architectural decisions about the Project CR which haven't been made yet.

Assuming a GitOps setup, consider the possible relationships between a GS Project CR and Flux. Does the Project CR live inside an application repository? If so, it would be a WC level resource. If the Project CR is managed separately from the application, it could be an MC-level resource. Those have tradeoffs and to my knowledge they haven't been discussed yet. 

So this mechanism could be WC-level (or could have complementary abstractions, like a label-based version of the same Project CR fields).

Reply:
So the issue I have is that in my mind, a project CR would need need to be in the MC because a project can be cross clusters especially if customers have a cluster per stage as this is often the case.

Reply:
Sure, maybe the Project CR ends up being MC-scoped, it just hasn't been decided. It definitely needs more wider discussion. My comment here was mostly to consider if there are aspects of logging that need to be under dev control, and if so, what that interface would look like.

Reply:
We might in any case need a leveled approach where we have project CR on MC level, and some other CR on WC levels (a project needs to be deployed to x WCs). But even that WC level in some cases lives on the MC, when you look at HelmRelease or App CR for example. Other parts of WC level would live within WC and that's most probably unavoidable. Thus, to me it means we can handle these independently. It depends a lot on the architecture of implemenation, which the con below shows. Handling some delegation from an MC level resource to a WC level one is not a big issue and we do that regularly with Apps anyway.